webdictionary/src/main/java/com/eis0/kademlianetwork/IdFinderHandler.java

```diff
@@ -1,2 +1,153 @@
 package com.eis0.kademlianetwork;
 
+import com.eis.smslibrary.SMSManager;
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+import com.eis0.kademlia.KademliaId;
+import com.eis0.kademlia.SMSKademliaNode;
+import com.eis0.kademlia.SMSKademliaRoutingTable;
+
+import java.math.BigInteger;
+
+/**
+ * This Class is used to find the closest node ID to a specific resource ID inside the network
+ *
+ * @author Marco Cognolato (wrote the class)
+ * @author Enrico Cestaro (edited and restored the code, to implement multiple research modes)
+ * @author Edoardo Raimondi (added the check cicle to verify if the target node is 'alive')
+ */
+public class IdFinderHandler {
+    /**
+     * Searches for a specific Id, which needs to be closer to a given resource Id.
+     * If it's found it's sent to a given SMSPeer who's searching, else sends a request to find it
+     * to a closer person than the actual node. By a chain process, if the node is not found
+     * it's sent to searcher the closest id found on the net
+     *
+     * @param idToFind     The ID to find in the network
+     * @param searcher     The Peer which is searching the ID
+     * @param researchMode The Resource mode, represents the final purpose of the research
+     * @throws IllegalArgumentException If the idToFind, the searcher or the researchMode are null
+     */
+    public static void searchId(KademliaId idToFind, SMSPeer searcher, ResearchMode researchMode) {
+        if (searcher == null || idToFind == null || researchMode == null)
+            throw new IllegalArgumentException();
```

Comment:
I think you are totally right, this could be a precious addition to my coding skills, especially for debugging; and yes, it's a problem in all my code, from now it will be a good habitude of mine doing it